### PR TITLE
Tegra changes from downstream v1

### PR DESCRIPTION
**Pull request by vwadekar**
Commits from the downstream Tegra port. These are not all of the changes as I want to get them merged in batches to avoid a lot of churn.

### CODE REVIEW

**Comment by vwadekar** [comment]
@danh-arm 

Hi Dan,

Please let me know if I should be pushing all changes in one go. There are more than 60 commits in the downstream branch.

File: include/lib/cpus/aarch64/denver.h [hunk 1]
```diff
@@ -41,6 +41,9 @@
 /* Implementer code in the MIDR register */
 #define DENVER_IMPL			0x4E
 
+/* Implementer code in the MIDR register */
+#define DENVER_IMPL			0x4E
```
Comment (davidcunado-arm) [inline include/lib/cpus/aarch64/denver.h]: Second define of DENVER_IMPL.

**Comment by vwadekar** [comment]
@davidcunado-arm 

Thanks. I removed the offending commit as it was not required and pushed the branch again.

**Comment by davidcunado-arm** [comment]
@vwadekar ok great. 
From my perspective, probably easier to review the changes in batches rather than a single huge PR!

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by davidcunado-arm** [comment]
@vwadekar build is failing for:
tegra-t132-trusty (Build options: CROSS_COMPILE=aarch64-linux-gnu- V=1 DEBUG=1 PLAT=tegra TARGET_SOC=t132 SPD=trusty)
./build/tegra/t132/debug/bl31/denver.o:(cpu_ops+0x0): undefined reference to `DENVER_1_0_MIDR'

tegra-t132-tlkd (Build options: CROSS_COMPILE=aarch64-linux-gnu- V=1 DEBUG=1 PLAT=tegra TARGET_SOC=t132 SPD=tlkd)
./build/tegra/t132/debug/bl31/denver.o:(cpu_ops+0x0): undefined reference to `DENVER_1_0_MIDR'

**Comment by vwadekar** [comment]
@davidcunado-arm 

My apologies. I have fixed the branch.

**Comment by vwadekar** [comment]
@davidcunado-arm 

Would I be able to fire a jenkins job before asking for reviews?

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by davidcunado-arm** [comment]
@vwadekar 
The expectation is that PRs are ready for review, and the Jenkins job should be a safety net to catch any unexpected impacts of a change.
Also, I wouldn't expect a failed jenkins job to result in a fundamental change in the contents of the PR and so should be 'safe' to review without having a jenkins job run.
No concerns with running Jenkins jobs multiple times!

**Comment by davidcunado-arm** [comment]
@vwadekar 
Fail was in static checks:
ERROR:7756b23:plat/nvidia/tegra/common/drivers/memctrl/memctrl_v1.c:System includes not in order.

**Comment by antonio-nino-diaz-arm** [comment]
Actually, the include order failure is a false positive, it seems to me that the current order is the correct one.

**Comment by davidcunado-arm** [comment]
Looks like that's the issue - mmio.h needs to be placed after memctrl_v1.h.

**Comment by vwadekar** [comment]
@antonio-nino-diaz-arm @davidcunado-arm 

Thanks. Pushed an updated branch

**Comment by danh-arm** [comment]
jenkins: retest this please

**Comment by davidcunado-arm** [comment]
@vwadekar 
Still getting a build error with the tegra132 builds: - e.g.

Build options: CROSS_COMPILE=aarch64-linux-gnu- V=1 DEBUG=1 PLAT=tegra TARGET_SOC=t132 SPD=trusty
./build/tegra/t132/debug/bl31/denver.o:(cpu_ops+0x0): undefined reference to `DENVER_1_0_MIDR'
make: *** [build/tegra/t132/debug/bl31/bl31.elf] Error 1

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Pull request merged**